service/history/replication/task_processor_manager.go

```diff
@@ -131,7 +134,9 @@ func (r *taskProcessorManagerImpl) Start() {
 	}
 
 	// Listen to cluster metadata and dynamically update replication processor for remote clusters.
-	r.listenToClusterMetadataChange()
+	if r.enableFetcher {
```

Comment:
with streaming disable, it should listen to metadata change as well?

Reply:
this is fetcher specific
there is also a stream specific var: https://github.com/temporalio/temporal/pull/4097/files#diff-9c489cc6ae52b9289979f47fcafaca2691ad9e7bf4e537597a4bbf68913b0217R42